dropshot/examples/basic.rs

```diff
@@ -58,15 +58,15 @@ async fn main() -> Result<(), String> {
     /*
      * Set up the server.
      */
-    let mut server = HttpServer::new(&config_dropshot, api, api_context, &log)
+    let server = HttpServer::new(&config_dropshot, api, api_context, &log)
         .map_err(|error| format!("failed to create server: {}", error))?;
     let server_task = server.run();
 
     /*
      * Wait for the server to stop.  Note that there's not any code to shut down
      * this server, so we should never get past this point.
      */
-    server.wait_for_shutdown(server_task).await
+    server_task.with_graceful_shutdown(std::future::pending()).await
```

Comment:
This feels pretty awkward to me, especially being the "hello world" golden path.  I suggested elsewhere just dropping the `with_graceful_shutdown()` function, in which case we could presumably just remove these calls.  It might be a problem if there's nothing here because the program might exit.  (I haven't confirmed this.)  I took a look at [hyper's "hello world" guide](https://hyper.rs/guides/server/hello-world/) and it looks like they give you a way to just wait for the server to exit without having to provide a never-resolving future argument.  That seems reasonably clean.  (In hyper, the server itself impls Future, which seems too overloaded to me, but we could just have a method that does this.)

This affects the other examples too.

More on this below.

Reply:
Ack, updated invocation is now:

```rust
server.await
```

(see below for details)